src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -47,6 +49,18 @@
 SET_EXCEPT = analyzer_utils.EXCEPT
 
 
+@dataclass
+class WhereClause:
+    expression: Expression
+    dependent_columns: Set[str]
```

Comment:
Should be `Optional[Set[str]]` since depdent_columns can be `None` when the Expression is not parsed (e.g. created by `sql_expr`)

Reply:
Should we make the change then?